lib/tutf8e/src/tutf8e.c

```diff
@@ -3,121 +3,200 @@
 
 #include <string.h>
 
+
+uint32_t tutf8e_encoder_flag(const char *string_flag) 
+{
+  if(string_flag == NULL || *string_flag == 0) {
+    return 0;
+  }
+  switch(*string_flag) {
+  case 'f':
+    if(!strcmp(string_flag,"fail")) {
+      return TUTF8E_FLAG_INV_FAIL;
+    }
+    break;
+
+  case 'i':
+    if(!strcmp(string_flag,"ignore")) {
+      return TUTF8E_FLAG_INV_IGNORE;
+    }
+    break;
+
+  case 'k':
+    if(!strcmp(string_flag,"keep")) {
+      return TUTF8E_FLAG_INV_KEEP;
+    }
+    break;
+  
+  case 'r':
+    if(!strcmp(string_flag,"replacement")) {
+      return TUTF8E_FLAG_INV_REPLACEMENT;
+    }
+    break;
+  
+  case 'q':
+    if(!strcmp(string_flag,"question")) {
+      return TUTF8E_FLAG_INV_QUESTION;
+    }
+    break;
+  default:
+    break;
+  }
+  return (uint32_t)(-1);
+}
+
 /* Determine the input length and UTF8 encoded length of NUL-terminated input string */
-/* return TUTF8E_INVALID if input character is not convertable                       */
-/* return TUTF8E_OK for success                                                      */
+/* return TUTF8E_INVALID if input character is not convertable  TUTF8E_FLAG_INV_FAIL */
+/* return TUTF8E_SAME    if string doesnt' change  (but copied) */
```

Comment:
Suggest deleting the words "but copied" here.   tutf8e_string_length does not copy.

Reply:
yeah... (as this is still WIP) comments needs to looking around.